### PR TITLE
upload external pattern lib smoke test

### PR DESCRIPTION
**Pull request by lyiyu66**

### CODE REVIEW

File: autotest/cldrpattern/test.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+
+
+@ddt
+class Test(unittest.TestCase):
+
+    @data(*excelutil())
+    def test(self, case):
+
+        if (case['casename'] == 'Check the number of folders in cldr folder'):
```
Comment (surite) [inline autotest/cldrpattern/test.py:45]: can you give a description about what these verification means? so people would know how you test it.

**Review by surite — CHANGES_REQUESTED** [review]

File: autotest/cldrpattern/test.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import os,json,HTMLTestRunner,unittest
+from excelutility import excelutil
+from ddt import ddt,data
+from datetime import datetime
+
+# Set testdata directory:
+# rootdir = 'D:\\PycharmProjects\\SmokeTesting\\singleton-i18n-patterns-core-0.5.1' #The path of cldr data storage
+# test_dir = 'D:\\PycharmProjects\\SmokeTesting' #The directory of generated test report
+rootdir = './singleton-i18n-patterns-core-0.5.1' #The path of cldr data storage
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/cldrpattern/test.py:9]: Strictly speaking, there is no version in external pattern lib branch. The rootdir will be updated once the branch and code structure is done.

File: autotest/cldrpattern/excelutility.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    datas = []
+
+    for row in sheet.iter_rows(min_row=2, values_only=True):
+        # data = TestData(casename=row[0],
+        #                 path=row[1],
+        #                 filename=row[2],
+        #                 key=row[3],
+        #                 expected=row[4])
+        data = {"caseid": row[0],
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/cldrpattern/excelutility.py]: Please see the latest version.

File: autotest/cldrpattern/excelutility.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from openpyxl import load_workbook
+
+def excelutil():
+    workbook = load_workbook(filename='ExternalPatternLibTestCases.xlsx')
```
[the remark this answers is not in the document]
Comment (lyiyu66) [reply autotest/cldrpattern/excelutility.py]: Adding a constant.py for storing some global settings. Please see the latest verison.

File: autotest/cldrpattern/test.py (hunk 1 above)
Comment (lyiyu66) [reply autotest/cldrpattern/test.py]: Adding case category to identify the case purpose, please see the latest version.

**Review by surite — APPROVED** [review]

**Pull request merged**